source/credentials_provider_imds.c

```diff
@@ -333,9 +437,27 @@ static void s_imds_on_stream_complete_fn(struct aws_http_stream *stream, int err
     struct aws_credentials_provider_imds_impl *impl = imds_user_data->imds_provider->impl;
     impl->function_table->aws_http_stream_release(stream);
 
+    // Token requst finished
+    if (imds_user_data->query_state == AWS_IMDS_QS_TOKEN) {
+        if (s_imds_on_token_response(user_data) != 0) {
+            AWS_LOGF_ERROR(AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to process IMDS token response.");
+            s_imds_finalize_get_credentials_query(imds_user_data);
+        }
+        return;
+    }
+
+    // Unauthorized for other requests, means service requires token.
+    if (imds_user_data->status_code == 401) {
```

Comment:
For status codes that aren't as well known, I would make a constant with both the number and the name in it (ala HTTP_STATUS_CODE_401_BLAH_BLAH)

Reply:
I wish these were defined in aws-c-http